examples/vision/reptile.py

```diff
@@ -0,0 +1,288 @@
+        temp_image *= 255
+        temp_image = np.clip(temp_image, 0, 255).astype("uint8")
+        if b == 2:
+            axarr[a, b].set_title("Class : " + sample_keys[a])
+        axarr[a, b].imshow(temp_image, cmap="gray")
+        axarr[a, b].xaxis.set_visible(False)
+        axarr[a, b].yaxis.set_visible(False)
+plt.show()
+
+"""
+## Build the model
+"""
+
+
+def conv_bn(x):
+    x = layers.Conv2D(filters=64, kernel_size=3, strides=2, padding="same")(x)
+    x = layers.BatchNormalization()(x)
+    return layers.ReLU()(x)
+
+
+inputs = layers.Input(shape=(28, 28, 1))
+x = conv_bn(inputs)
+x = conv_bn(x)
+x = conv_bn(x)
+x = conv_bn(x)
+x = layers.Flatten()(x)
+outputs = layers.Dense(classes, activation="softmax")(x)
+model = keras.Model(inputs=inputs, outputs=outputs)
+model.compile()
+optimizer = keras.optimizers.SGD(learning_rate=learning_rate)
+
+"""
+## Train the model
+"""
+
+training = []
+testing = []
+for meta_iter in range(meta_iters):
+    frac_done = meta_iter / meta_iters
+    cur_meta_step_size = (1 - frac_done) * meta_step_size
+    # Temporarily save the weights from the model.
+    old_vars = model.get_weights()
+    # Get a sample from the full dataset.
+    mini_dataset = train_dataset.get_mini_dataset(
+        inner_batch_size, inner_iters, train_shots, classes
+    )
+    for images, labels in mini_dataset:
+        with tf.GradientTape() as tape:
+            preds = model(images)
+            loss = keras.losses.sparse_categorical_crossentropy(labels, preds)
+        grads = tape.gradient(loss, model.trainable_weights)
+        optimizer.apply_gradients(zip(grads, model.trainable_weights))
+    new_vars = model.get_weights()
+    # Perform SGD for the meta step.
+    for var in range(len(new_vars)):
+        new_vars[var] = old_vars[var] + (
+            (new_vars[var] - old_vars[var]) * cur_meta_step_size
+        )
+    # After the meta-learning step, reload the newly-trained weights into the model.
+    model.set_weights(new_vars)
+    # Evaluation loop
+    if meta_iter % eval_interval == 0:
+        accuracies = []
+        for dataset in (train_dataset, test_dataset):
+            # Sample a mini dataset from the full dataset.
+            train_set, test_images, test_labels = dataset.get_mini_dataset(
+                eval_batch_size, eval_iters, shots, classes, split=True
+            )
+            old_vars = model.get_weights()
+            # Train on the samples and get the resulting accuracies.
+            for images, labels in train_set:
+                with tf.GradientTape() as tape:
+                    preds = model(images)
+                    loss = keras.losses.sparse_categorical_crossentropy(labels, preds)
+                grads = tape.gradient(loss, model.trainable_weights)
+                optimizer.apply_gradients(zip(grads, model.trainable_weights))
+            test_preds = model.predict(test_images)
+            test_preds = tf.argmax(test_preds).numpy()
```

Comment:
@ADMoreau Shall we add `axis=1` to `tf.argmax` here, so that we pick the largest index from different class predictions? Otherwise, we're picking the largest index from the batch direction.